docs/04-reference/winglang-spec.md

```diff
@@ -2062,11 +2064,52 @@ let bucket = cdk.aws_s3.Bucket(
 );
 ```
 
-### 5.2 Internal Libraries
+### 5.1.2 Internal Libraries
 
 Wing libraries themselves are JSII modules. They can be used in all other JSII
 supported languages.
 
+## 5.2 JavaScript
+
+The `extern(file)` modifier can be used on method declarations (in classes and resources) to indicate that a method is backed by an implementation imported from a JavaScript file.
+
+In the following example, the static inflight method `make_id` is implemented
+in `helper.js`:
+
+```js
+// task-list.w
+resource TaskList {
+  // ...
+
+  inflight add_task(title: str) {
+    let id = TaskList.uuid();
+    this.bucket.put(id, title);
+  }
+
+  extern("./helpers.js") static inflight make_id(): str;
```

Comment:
What about:

```
inflight foo(x: str): void = extern "./helper.js";
```

Reply:
Then that would look like an expression, in which case it would actually be more consistent to treat extern like a global function and call it with extern("./helper.js")

Reply:
So:

```js
inflight foo(x: str): void = extern("./helper.js");
```

Reply:
I am okay with this although it feels like it would make sense to bind "extern" and "helper.js" somehow, which is why I used the call notation.

```js
extern "helper.js" inflight foo(x: str): void;
```

> `extern "helper.js" { .. }`

This implies that now `foo` is a global function, which we currently don't support.

Reply:
Updated!